log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/ResourceUtil.java

```diff
@@ -81,6 +84,22 @@ private static String loadClasspathResource(final String path) {
         }
     }
 
+    private static InputStream loadClasspathResource(String resourcePath, ClassLoader... classLoaders) {
+        for (ClassLoader classLoader : classLoaders) {
+            final InputStream resource = classLoader.getResourceAsStream(resourcePath);
+            if (resource != null) {
+                return resource;
+            }
+        }
+        return null;
+    }
+
+    private static ClassLoader[] getClassLoaders() {
+        return Stream.of(ClassLoader.getSystemClassLoader(), Thread.currentThread().getContextClassLoader())
```

Comment:
Even though it's lazy evaluated, don't you still want to suppress the possible SecurityException from Thread.currentThread().getContextClassLoader()?

Why did you remove the ResourceUtil.class.getClassLoader(), do you know its always the same as one of the others?

Reply:
> do you know its always the same as one of the others

In case of Spring Boot - yes (assuming that hash codes don't lie). Feel free to retest.

As for suppression - no. If `java.security` policy doesn't allow it, it should scream as loud as possible. Suppressing it will blur the picture unnecessarily - "Requested classpath resource was null" would hide the real cause. Also, if it's not thrown [here](https://github.com/rfoltyns/log4j2-elasticsearch/blob/8cbc3c6d5d548839481fc9e8fa211723edb1e64d/log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/spi/BatchEmitterServiceProvider.java#L47) it's all good anyway.

Reply:
Agree, `java.security` should scream as loud as possible. But in the case security is enabled, and `Thread.currentThread().getContextClassLoader()` is not allowed, I think the resource loading will always fail with a `SecurityException`. Isn't that correct?

Reply:
Isn't that the purpose of `java.security`? :slightly_smiling_face:

Reply:
Sure, but in this case i think it will be alright to load the resources with the other class loaders even though access to `Thread.currentThread().getContextClassLoader()` is prohibited. 🙂

Reply:
> i think it will be alright to load the resources with the other class loaders even though access to Thread.currentThread().getContextClassLoader() is prohibited

Very good point. But, as mentioned before, it should fail [earlier](https://github.com/rfoltyns/log4j2-elasticsearch/blob/8cbc3c6d5d548839481fc9e8fa211723edb1e64d/log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/spi/BatchEmitterServiceProvider.java#L47). It was requested by OSGi users and if `java.security` prevents it, it's a setup issue outside of scope of this repo.

Reply:
Any more thoughts on this? I'm happy to commit it and release it as-is once squashed.

Reply:
Sorry for the late reply. 
No, if the intention is to fail fast, then everything looks good to me.